.github/workflows/modern-distributions.yml

```diff
@@ -0,0 +1,34 @@
+name: Validate tar based distributions
+
+on:
+  pull_request:
+    paths: ['distributions/**']
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check:
+    name: Validate tar based distributions changes
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout repo
+          uses: actions/checkout@v4
+          with:           
+            fetch-depth: 0
+
+
+        - name: Install pip packages
+          run: pip install -r distributions/requirements.txt
+          shell: bash
+
+        - name: Run validation
+          run: | 
+            python distributions/validate-modern.py \
+            --repo-path . \
+            --compare-with-branch 'origin/${{ github.base_ref }}' \
+            --manifest distributions/DistributionInfo.json \
+            --github-token '${{ secrets.GITHUB_TOKEN }}' \
```

Comment:
How is this token managed?

Reply:
It's a token that's generating by github actions. We don't actively have to manage it (https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication)